ctapipe/tools/process.py

```diff
@@ -109,16 +115,37 @@ def setup(self):
                 "shower distributions read from the input Simulation file are invalid)."
             )
 
+    @property
+    def should_compute_dl2(self):
+        """ returns true if we should compute DL2 info """
+        return self.write.write_stereo_shower or self.write.write_mono_shower
+
+    @property
+    def should_compute_dl1(self):
+        """returns true if we should compute DL1 info"""
+        return self.write.write_parameters or self.should_compute_dl2
+
     def _write_processing_statistics(self):
         """ write out the event selection stats, etc. """
         # NOTE: don't remove this, not part of DataWriter
-        image_stats = self.process_images.check_image.to_table(functions=True)
-        image_stats.write(
-            self.write_dl1.output_path,
-            path="/dl1/service/image_statistics",
-            append=True,
-            serialize_meta=True,
-        )
+
+        if self.should_compute_dl1:
+            image_stats = self.process_images.check_image.to_table(functions=True)
+            image_stats.write(
+                self.write.output_path,
+                path="/dl1/service/image_statistics",
+                append=True,
+                serialize_meta=True,
+            )
+
+        if self.should_compute_dl2:
+            shower_stats = self.process_shower.check_shower.to_table(functions=True)
+            shower_stats.write(
+                self.write.output_path,
+                path="/dl2/service/image_statistics",
```

Comment:
shower_statistics?

Reply:
it's not really shower statistics, it's still cuts on each telescope image/parameters that are later used to reconstruct the shower .  I thought about renaming both to cut_statistics, in fact, but decided not to change the data model.

Reply:
> but decided not to change the data model.

THis is a new field. so what would a different name change?

Reply:
no I mean changing both.  In the end I decided to keep the old name for both, rather than naming both to "cut_statistics".

Reply:
both are statistics on images used for X, where X is parameterization (DL1) or reconstruction (DL2).  For a future PR/issue, these are not super useful right now, and would benefit from being broken down by tel_id or something.